tasks/README.md

```diff
@@ -29,22 +29,19 @@ Tasks are packaged and distributed in the `/tasks` directory of a Puppet module.
 
 ### Task name and filename
 
-- Task names have the same restriction as puppet type names and must match the regular expression `\A[a-z][a-z0-9\_]*\z`.
-- Filenames not matching the task name regular expression will be ignored.
-- Tasks are referred to as `module_name::task_name`.
-- Tasks must exist at the top level of the tasks directory to be found.
-- Task metadata is stored in `task_name.json` and is optional.
-- The executable for the task must be stored in `task_name` or `task_name.<extension>`(where extension != `json`, `md` or `conf`).
-- The presence or absence of an execute bit should be ignored by the task runner when finding task files and metadata.
-- The name `init` and `init.<extension>` are treated specially and the init task may be referred to by the shorthand `module_name`.
+A task consists of an optional metadata file and one or more implementation files. Files with the `.json` extensions are metadata files, and any other file extension (including files with no extension) is an implementation file. Implementation files do not need the executable bit set.
 
-A task with multiple executable files will be considered invalid.
+An implementation file `<task>.<ext>` without a corresponding metadata file `<task>.json` is a task.
 
-Tools for validating this metadata and making local task execution easier when testing will eventually be added to the PDK.
+A metadata file `<task>.json` also identifies a task. If the metadata specifies an `implementations` array, then the files listed in that array are the implementations for the task. Otherwise, there must be one corresponding implementation file `<task>.<ext>`.
+
+Task names have the same restriction as puppet type names and must match the regular expression `\A[a-z][a-z0-9\_]*\z`. The extensions `.md` and `.conf` are forbidden. Only files at the top level of the `tasks` directory matching the task name regular expression are used; all other files are ignored.
```

Comment:
Are there any other constraints on the `.<ext>`, other than "not `.md` and `.conf`"?

Reply:
not .json